src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntity.php

```diff
@@ -55,7 +57,7 @@ public function __construct(
         string $entityClass = null,
         string $repositoryMethod = null,
         string $errorPath = null,
-        bool $ignoreNull = null,
+        bool|string|array $ignoreNull = null,
```

Comment:
You shouldcheck whether `ignoreNull` value(s) are defined within `fields` and probably throw an exception if that's not the case

Reply:
Where should I add the check @malarzm ?

My first thought was inside the `construct` but, if you look at the validator, the check
```
if (!\is_array($constraint->fields) && !\is_string($constraint->fields)) {
    throw new UnexpectedTypeException($constraint->fields, 'array');
}
```
is inside the validate method.
So technically, in the constraint construct, I cannot consider that the `fields` property will always be a string/array, no ?

Reply:
Might. be worth waiting for a core-member opinion about where checking this then :)